design/ceph/external-cluster-monitoring.md

```diff
@@ -0,0 +1,63 @@
+---
+title: external-cluster-monitoring
+target-version: release-1.4
+---
+
+# External cluster monitoring
+
+## Summary
+
+Rook already has the ability to connect to an external Ceph cluster.
+When configured like so, Rook will consume resources from the external cluster.
+
+## Motivation
+
+In addition to what is possible today, we would like to provide monitoring functionality coming from Kubernetes to an external Ceph cluster.
+Rook already creates and injects service monitoring configuration, consuming what the ceph-mgr prometheus exporter module generates.
+
+### Goals
+
+RGW monitoring and alerting will be plugged into the Kubernetes monitoring and alerting framework.
+This enables the capability of a Kubernetes cluster to gather metrics from the external cluster and feed them in Prometheus.
+
+### Non-Goals
+
+Network connection of the external gateways in Rook.
+This assumes that all the networking plumbing has been done and connectivity is established between the Kubernetes cluster and the external Ceph cluster.
+
+## Proposal
+
+The idea is to allow Rook-Ceph to connect to an external ceph-mgr prometheus module exporter.
+
+1. Enhance external cluster script:
+    1. the script tries to discover the list of managers IP addresses
+    2. if provided by the user, the list of ceph-mgr IPs in the script is accepted via the new `--prometheus-exporter-endpoint` flag
+
+2. Add a new entry in the monitoring spec of the `CephCluster` CR:
+
+```go
+// Endpoint points to an existing Ceph prometheus exporter endpoint
+Endpoints []v1.EndpointAddress `json:"endpoints,omitempty"`
+}
+```
+
+So the CephCluster CR will look like:
+
+```yaml
+monitoring:
+  # requires Prometheus to be pre-installed
+  enabled: false
+  endpoints:
+    - ip: "192.168.0.2"
+    - ip: "192.168.0.3"
```

Comment:
it is very common for the port to change as well, so don't forget the port number here

Reply:
To change the port of the Prometheus exporter? Really?